### PR TITLE
[AI-FSSDK] [FSSDK-12337] Add Feature Rollout support

### PR DESCRIPTION
**Pull request by FarhanAnjum-opti**
## Summary

Adds Feature Rollout support to the Swift SDK. Feature Rollouts are a new experiment rule type that combines Targeted Delivery simplicity with A/B test measurement capabilities. During project config parsing, the "everyone else" variation from the flag's rollout is injected into any experiment with type "fr", enabling correct evaluation without changes to decision logic.

## Changes

- Added optional `type` field (ExperimentType enum) to the Experiment model with valid values: ab, mab, cmab, td, fr
- Added config parsing logic to inject the "everyone else" rollout variation into feature rollout experiments
- Added traffic allocation entry (endOfRange=10000) for the injected variation
- Added `getEveryoneElseVariation` helper method to extract the last rollout rule's first variation
- Updated experiment lookup maps (experimentKeyMap, experimentIdMap) after injection so decisions use updated data
- Added 10 unit tests covering feature rollout injection, edge cases, and backward compatibility

## Jira Ticket

[FSSDK-12337](https://optimizely-ext.atlassian.net/browse/FSSDK-12337)

[FSSDK-12337]: https://optimizely-ext.atlassian.net/browse/FSSDK-12337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Sources/Data Model/Experiment.swift [hunk 1]
```diff
@@ -74,4 +85,8 @@ extension Experiment {
     var isCmab: Bool {
         return cmab != nil
```
Comment (muzahidul-opti) [inline Sources/Data Model/Experiment.swift:102]: Can we update the logic as we did for feature rollout?

File: Sources/Data Model/ProjectConfig.swift [hunk 2]
```diff
@@ -135,7 +135,24 @@ class ProjectConfig {
...
+
+        // Feature Rollout injection: for each feature flag, inject the "everyone else"
+        // variation into any experiment with type == .featureRollout
+        injectFeatureRolloutVariations()
+
+        // Rebuild experiment maps after injection so lookup maps contain injected variations
+        self.experimentKeyMap = {
+            var map = [String: Experiment]()
+            allExperiments.forEach { map[$0.key] = $0 }
+            return map
```
Comment (muzahidul-opti) [inline Sources/Data Model/ProjectConfig.swift]: Why redundant rebuild? We may move the `injectFeatureRolloutVariations()` to the top before existing lookup building logic.

**Review by muzahidul-opti — COMMENTED** [review]
Looks good to me. A couple of comments added.

File: Sources/Data Model/Experiment.swift (hunk 1 above)
Comment (muzahidul-opti) [reply Sources/Data Model/Experiment.swift:102]: I think we need for backward compatibility

File: Sources/Data Model/ProjectConfig.swift (hunk 2 above)
Comment (muzahidul-opti) [reply Sources/Data Model/ProjectConfig.swift]: Remove redundant lookup logic
